integration-tests/script/copy_resources_template.sh

```diff
@@ -27,9 +27,16 @@ cp -r client_tls docker/client_tls
 rm -rf $SHARED_DIR/docker
 mkdir -p $SHARED_DIR
 cp -R docker $SHARED_DIR/docker
-mvn -B dependency:copy-dependencies -DoutputDirectory=$SHARED_DIR/docker/lib
 
-# Make directories if they dont exist
+pushd ../
+rm -rf apache-druid-$DRUID_VERSION
+mvn -DskipTests -T1C -Danimal.sniffer.skip=true -Dcheckstyle.skip=true -Ddruid.console.skip=true -Denforcer.skip=true -Dforbiddenapis.skip=true -Dmaven.javadoc.skip=true -Dpmd.skip=true -Dspotbugs.skip=true install -Pdist
+tar xzf distribution/target/apache-druid-$DRUID_VERSION-bin.tar.gz
```

Comment:
it looks like maven assembly plugin supports a 'dir' format, which "Creates an exploded directory format" according to the docs here https://maven.apache.org/plugins/maven-assembly-plugin/assembly.html. What do you think about making a custom integration-tests version of https://github.com/apache/druid/blob/master/distribution/src/assembly/assembly.xml (which leaves out all of the normal distribution stuff like examples and licenses and stuff) along with custom distribution build profile so we could skip making and then unpacking the archive? I suspect it would shave off some overhead, anecdotally it seems like this part is pretty slow when i do these things locally.

semi related, (but not to this PR) the normal docker image could probably stand to do this too https://github.com/apache/druid/blob/master/distribution/docker/Dockerfile#L43, but it would need all the normal distribution stuff so it would just need a way to override the format or just like make the 'dir' version the base and then a 'tar' version that depends on the output of the base version to archive it for a distribution.

Reply:
That's a good idea, though the extra time that this PR added in each integration test was about 2-5 min in travis which seems not that bad. It sounds better since it would be clearer. I added a new maven profile, `integration-test`, that creates an exploded directory under `distribution/target/`. Integration tests now use this new profile. In a simple test in my machine, the `integration-test` profile took 54 sec while the `dist` profile took 90 sec.